### PR TITLE
 fix time problem

### PR DESCRIPTION
**Pull request by henrycjchen**
## PR Type
- [ ] Snippets, Tests & Tags (new snippets, updated snippets, re-tagging of snippets, added/updated tests)
- [x] Scripts & Website & Meta (anything related to files in the [scripts folder](https://github.com/30-seconds/30-seconds-of-code/tree/master/scripts), how the repository's automated procedures work and the website)
- [ ] Glossary & Secondary Features (anything related to the glossary, such as new or updated terms or other secondary features)
- [ ] General, Typos, Misc. & Meta (everything else, typos, general stuff and meta files in the repository - e.g. the issue template)

## Guidelines
- [x] I have read the guidelines in the [CONTRIBUTING](https://github.com/30-seconds/30-seconds-of-code/blob/master/CONTRIBUTING.md) document.
- [x] My PR doesn't include any `testlog` changes.

### CODE REVIEW

**Comment by henrycjchen** [comment]
fix time problem

**Comment by henrycjchen** [comment]
fix time problem

**Comment by fejes713** [comment]
Could you please let us know what the problem is? 😄 Thanks

**Comment by henrycjchen** [comment]
I found this problem when using throttle in some Android's webview. And I've noticed that there is a discussion in stackoverflow about this setTimeout problem. (https://stackoverflow.com/questions/8430966/is-calling-settimeout-with-a-negative-delay-ok/8431015#8431015)

**Comment by atomiks** [comment]
You can do

```js
Math.max(0, wait - (Date.now() - lastTime))
```

File: snippets/throttle.md [hunk 1]
```diff
@@ -19,12 +19,14 @@ const throttle = (fn, wait) => {
...
       clearTimeout(lastFn);
+      let time = 0;
       lastFn = setTimeout(function() {
         if (Date.now() - lastTime >= wait) {
           fn.apply(context, args);
           lastTime = Date.now();
         }
-      }, wait - (Date.now() - lastTime));
+        time = wait - (Date.now() - lastTime)
+      }, time>0 ? time : 0);
```
Comment (atomiks) [inline snippets/throttle.md]: You can do

```js
Math.max(0, wait - (Date.now() - lastTime))
```

Comment (henrycjchen) [reply snippets/throttle.md]: agree

**Pull request merged**